flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTableUtil.java

```diff
@@ -403,6 +404,25 @@ public static Table instantiateHiveTable(ObjectPath tablePath, CatalogBaseTable
 				.collect(Collectors.toMap(t -> t.f0, t -> t.f1));
 	}
 
+	/**
+	 * Check whether to read or write on the hive ACID table.
+	 *
+	 * @param catalogTable Hive catalog table.
+	 * @param tablePath    Identifier table path.
+	 * @throws FlinkHiveException Thrown, if the source or sink table is transactional.
+	 */
+	public static void checkAcidTable(CatalogTable catalogTable, ObjectPath tablePath) {
+		if (catalogTable != null && catalogTable.getOptions() != null) {
```

Comment:
`catalogTable` and `tablePath` never null, you can take a look to the constructor of `HiveTableSource`

Reply:
@JingsongLi `checkAcidTable` is the util method for `HiveTableUtil`, which should add check for the parameter `catalogTable`.

Reply:
If really want to have a check, check should be `Preconditions.check`

Reply:
@JingsongLi Yeah, you are right. I remove this check for `catalogTable` with options.